packages/terra-tabs/src/Tabs.jsx

```diff
@@ -119,16 +135,40 @@ class Tabs extends React.Component {
           showIcon={child.props.showIcon}
           render={() => tabContent}
           isDisabled={child.props.isDisabled}
+          isClosable={this.props.isClosable !== undefined ? this.props.isClosable : false}
           variant="framework"
         />,
       );
     });
+    const handleTabsStateChange = (newValue, itemKey, event) => {
+      if (newValue.length > 0) {
+        let activeAfterClosed = '';
+        for (let i = 0; i < newValue.length; i += 1) {
+          if (newValue[i].isSelected === true) {
+            activeAfterClosed = newValue[i].itemKey;
+            break;
+          }
+        }
+        this.setState({ activeAfterClosed });
+      } else if (newValue.length === 0) {
+        this.setState({ activeAfterClosed: '' });
+      }
+      return this.props.onTabClose && this.props.onTabClose(newValue, itemKey, event);
+    };
 
     return (
       <CommonTabs
         id={customProps.id}
         activeItemKey={this.state.activeKey}
         onRequestActivate={key => this.setState({ activeKey: key })}
+        onClosingkey={(key) => {
+          if (this.state.activeKey === key) {
+            setTimeout(() => {
+              this.setState((prevState) => ({ activeKey: prevState.activeAfterClosed }));
+            }, 100);
+          }
+        }}
+        onClosingTab={handleTabsStateChange}
```

Comment:
Why do you need 2 different props to be triggered on closing the tab?

Reply:
The **onClosingTab** function determines the active selected tab in the tabs .
The activeAfterClosed state variable holds the key of the tab to become active after closure of activetab.
When in the case of Active tab is closed:The **onClosingkey** handle the case and updates the activeKey state according to requirement.

Reply:
Yes but are called on closing the tab right so why cannot we have one callback instead of 2.

Reply:
Right Saket , have refactor and made the changes https://github.com/cerner/terra-framework/commit/ca07c6755741f69957198bcb28e0ed3fbc6ffb15